src/neuroconv/tools/nwb_helpers/_configure_backend.py

```diff
@@ -0,0 +1,24 @@
+"""Collection of helper functions related to configuration of datasets dependent on backend."""
+from typing import Union
+
+from pynwb import NWBFile
+
+from ._configuration_models._hdf5_backend import HDF5BackendConfiguration
+from ._configuration_models._zarr_backend import ZarrBackendConfiguration
+
+
+def configure_backend(
+    nwbfile: NWBFile, backend_configuration: Union[HDF5BackendConfiguration, ZarrBackendConfiguration]
+) -> None:
+    """Configure all datasets specified in the `backend_configuration` with their appropriate DataIO and options."""
+    nwbfile_objects = nwbfile.objects
+
+    data_io_class = backend_configuration.data_io_class
+    for dataset_configuration in backend_configuration.dataset_configurations.values():
+        object_id = dataset_configuration.dataset_info.object_id
+        dataset_name = dataset_configuration.dataset_info.dataset_name
+        data_io_kwargs = dataset_configuration.get_data_io_kwargs()
+
+        # TODO: update buffer shape in iterator, if present
+
+        nwbfile_objects[object_id].set_data_io(dataset_name=dataset_name, data_io_class=data_io_class, **data_io_kwargs)
```

Comment:
It would be nice if we could get the neurodata object by location.

I am also curios on why the `set_data_io` is at the neurodata object level and not at the data datset level. That's where the IO stuff ends up right?

Reply:
> Disagree in this case

You disagree that it would be good to have it?

It is good to have both. Most people are used to file systems so that's easy to use but unique identifiers are important for validation and computer use. What would be the downside?

 In hdf5 you can access stuff by its path within the file which I think makes sense from an usability perspective.

Reply:
If you're asking/commenting in general that 'it would be nice to fetch an object in an NWB file knowing its path' then that already exists in the PyNWB API (it's that natural way of navigating group/dataset contents across the modules and submodules; but relies on closing each level of nesting)

But I think what you refer to is passing a single string that includes slashes to indicate the nested group structure (which you can also do in PyNWB; just pass it to the `nwbfile._file` attribute) but the location structure in the HDF5 file is not necessarily equivalent to the PyNWB module structure at all times. There are various aliases in the API that break that equivalency and even my 'location in file' isn't exactly either approach, nor is it intended to be

ALso this specific lookup is a piece of internal code masking what is really going on, which is reliance on object ID; what you seem to be asking for is a more generic features of PyNWB so I'd suggest just raise an issue over there to discuss with that team. I'm just saying it's not a feature I would ever use myself

Note that the 'location in file' used in the `BackendConfiguration` lookup dictionary here is only a convenient association to the object ID; is not 100% guaranteed to be perfectly identical to the PyNWB nesting structure at all times for all files, just 'generally close enough to distinguish references to contents'

That is, I've made this nice heuristic for mapping references of `location_in_file` -> `object_id` and then this line of internal operation, I go from `object_id` -> `neurodata_object`

Reply:
> But I think what you refer to is passing a single string that includes slashes to indicate the nested group structure 

Yes, exactly, that's what I was saying as a very later comment (not asking you) that it would be nice to have. For def I did not make it clear enough that is not related to this PR or even neuroconv. Sorry for making you write something larger than you would have had that context been clear.

> (which you can also do in PyNWB; just pass it to the nwbfile._file attribute) 
Really? I need something like this. This does not seem to work:
```
nwbfile._file["acquisition/ElectricalSeriesAp"]
AttributeError: 'NWBFile' object has no attribute '_file'
```

Reply:
And in the hdf5 case when streaming it i way cheaper to just know the specific location.